hail/python/hailtop/hailctl/dataproc/modify.py

```diff
@@ -39,32 +40,20 @@ def main(args, pass_through_args):
         modify_args.append('--max-idle={}'.format(args.max_idle))
 
     if modify_args:
-        cmd = ['gcloud',
-               'dataproc',
-               'clusters',
-               'update',
-               args.name] + modify_args
+        cmd = ['dataproc', 'clusters', 'update', args.name] + modify_args
 
         if args.beta:
-            cmd.insert(1, 'beta')
+            cmd.insert(0, 'beta')
 
         cmd.extend(pass_through_args)
 
         # print underlying gcloud command
-        print(' '.join(cmd[:5]) + ' \\\n    ' + ' \\\n    '.join(cmd[5:]))
+        print('gcloud ' + ' '.join(cmd[:4]) + ' \\\n    ' + ' \\\n    '.join(cmd[5:]))
 
         # Update cluster
         if not args.dry_run:
             print("Updating cluster '{}'...".format(args.name))
-            sp.check_call(cmd)
-
-    if args.zone:
-        zone = args.zone
-    else:
-        zone = sp.check_output(["gcloud", "config", "get-value", "compute/zone"], stderr=sp.DEVNULL).decode().strip()
-
-    if not zone:
-        raise RuntimeError("Could not determine compute zone. Use --zone argument to hailctl, or use `gcloud config set compute/zone <my-zone>` to set a default.")
```

Comment:
Zone is only required when modifying the Hail wheel on the cluster. With this check here, `hailctl dataproc modify some-cluster --num-workers=2` would unnecessarily error if no zone was provided or `compute/zone` was not set.

Reply:
ah, didn't know that!

Reply:
It's the `gcloud compute` commands that take a zone argument. The `gcloud dataproc` commands for working with an existing cluster only use the region.

I recently noticed that `hailctl dataproc diagnose` handles this by using `gcloud dataproc clusters describe` to obtain the zone for a cluster.
https://github.com/hail-is/hail/blob/71801b763ea2653fc6a6fcec812c812f57fe9a7e/hail/python/hailtop/hailctl/dataproc/diagnose.py#L39-L51